File: task/src/main/java/com/goexp/galgame/data/piplline/handler/DefaultStarter.java
package com.goexp.galgame.data.piplline.handler;

import com.goexp.galgame.data.piplline.core.Message;
import com.goexp.galgame.data.piplline.core.MessageQueueProxy;

import java.util.concurrent.BlockingQueue;

public abstract class DefaultStarter<T> extends DefaultMessageHandler<T> implements Starter {

    @Override
    public void process(Message<T> message, MessageQueueProxy<Message> msgQueue) {
        process(msgQueue);
    }

}

File: task/src/main/scala/com/goexp/galgame/data/piplline/handler/DefaultStarter.scala
package com.goexp.galgame.data.piplline.handler

import com.goexp.galgame.data.piplline.core.{Message, MessageQueueProxy}

abstract class DefaultStarter extends Starter {
  var queue: MessageQueueProxy[Message[_]] = _

  def setQueue(queue: MessageQueueProxy[Message[_]]): Unit = this.queue = queue

  def send(mes: Message[_]) = queue.offer(mes)
}